File: ruby/arcade/addBorder.rb
def addBorder(picture)
	r, c = [], 0
	(0...(picture.size + 2)).each do |t|
		if t == 0 or t == (picture.size + 1) then
			r[t] = "*"*(picture[0].size + 2)
			next
		end
		r[t] = "*#{picture[c]}*"
		c += 1
	end
	return r
end

File: java/arcade/AddBorder.java
public final class AddBorder {

	String[] addBorder(String[] picture) {
		String b = "";
		int len = picture.length + 2;
		String[] r = new String[len];
		for(int i = 0; i < picture[0].length() + 2; i++) { b += "*"; }
		for(int i = 0, x = 0; i < len; i++) {
			if(i == 0 || i == len - 1) {
				r[i] = b;
				continue;
			}
			r[i] = "*"+picture[x]+"*";
			x++; 
		}
		return r;
	}
}
